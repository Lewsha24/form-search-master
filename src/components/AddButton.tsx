type Props = {
    onClick: () => void;
}

export const AddButton = ({onClick}: Props) => {
    return (
        <button onClick={onClick}>Add</button>
    )
}