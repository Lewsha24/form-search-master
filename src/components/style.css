.input-field {
    display: flex;
    flex-flow: column nowrap;

    width: 300px;
    padding: 24px;

}

.input-field label {
    font-size: 12px;
    color: darkgray;
}