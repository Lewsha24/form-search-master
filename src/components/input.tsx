import {ChangeEvent, useState} from "react";
import "../components/style.css"
type Props = {
    defaultValue: string,
    onChange: (value: string) => void,
}

// В нипут передаем значение и текст
// value - значение
// onChange - Передаем event handler
export const Input = ({defaultValue, onChange}: Props) => {
    // lableValue - хранит выводимую ошибку
    const [lableValue, setLableValue] = useState("")
    const [inputValue, setInputValue] = useState(defaultValue)
    // функция Которая вызывается
    const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
        // Принимаем значение из поля input
        const val = event.target.value;
        // Делаем проверку Количества символов
        if(val.length > 32) {
            setLableValue(`Привышена длина заголовка ${val.length}`)
        }else {
            setLableValue("")
            setInputValue(val)
            // Передаем в функцию значение val
            onChange(val)
        }
    }
    return (
        <div className="input-field">
            <label
                htmlFor="input-header-field-id"
                data-testid="input-lable"
            >Введите заголовок</label>
            <input
                id="input-header-field-id"
                placeholder={`например, введите "купите молоко"`}
                type="text"
                aria-hidden={false}
                alt="Поле ввода заголовка задачи"
                title="Поле ввода заголовка"
                onChange={handleChange}
                value={defaultValue}
                data-testid="input-field"
            />
            <span data-testid="hint-text">{lableValue}</span>
        </div>
    )
}