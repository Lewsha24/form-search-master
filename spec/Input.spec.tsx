import {screen, render,} from "@testing-library/react";
import {Input} from "../src/components/input";
import uE from "@testing-library/user-event";
import '@testing-library/jest-dom';
describe("Поле ввода", () => {

    const userEvent = uE.setup({delay: 100});

    // it.todo("огранечение на ввод более 32 символов");

    it("Поле ввода", async () => {
        const fn = jest.fn();

        render(<Input defaultValue="" onChange={fn}/>);
        // Находим элемент
        const input = screen.getByRole("textbox");
        await userEvent.type(input, 'Hello')
        await userEvent.tripleClick(input)
        const cuttedValue = await userEvent.cut()

        await userEvent.clear(input);

        await userEvent.paste(cuttedValue)

        expect(screen.getByDisplayValue(/Hello/i)).toBeInTheDocument;
    });
});
