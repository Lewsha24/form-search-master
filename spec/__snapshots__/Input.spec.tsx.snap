// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Поле ввода Поле доступен для воода 1`] = `
<DocumentFragment>
  <div
    class="input-field"
  >
    <span>
      Input
    </span>
    <input
      type="text"
      value="Greetings!"
    />
    <label />
  </div>
</DocumentFragment>
`;
